File: clients/rn/src/engine/persistence/VitalityStore.ts
/**
 * Persistence for agent vitality state.
 * TypeScript port of KMP's VitalityStore.kt.
 */

import type { VitalityState } from '../agent/VitalityState';

export interface VitalityStore {
  save(entityId: string, state: VitalityState): Promise<void>;
  load(entityId: string): Promise<VitalityState | null>;
}

File: clients/kmp/shared/src/commonMain/kotlin/org/wyrdsekai/app/engine/persistence/VitalityStore.kt
package org.wyrdsekai.app.engine.persistence

import org.wyrdsekai.app.engine.agent.VitalityState

/**
 * Persistence for agent vitality state.
 * Platform-specific implementations: SQLite on iOS, stubs elsewhere.
 */
interface VitalityStore {
    suspend fun save(entityId: String, state: VitalityState)
    suspend fun load(entityId: String): VitalityState?
}
